src/app/edit-collection/edit-collection.component.html

```diff
@@ -31,37 +32,36 @@
   <span ngbDropdown>
     <button id="single-button" type="button" ngbDropdownToggle mat-raised-button class="btn btn-success btn-xs"
       [disabled]="disabled || saving">
-      Version {{version.number}} &nbsp;&nbsp;&#8208;&nbsp;&nbsp; {{version.creationDate | date:'short'}}&nbsp;&nbsp;
+      Version {{version.number}} &nbsp;&nbsp;&#8208;&nbsp;&nbsp; {{version.name}} &nbsp;&nbsp;&#8208;&nbsp;&nbsp; {{version.creationDate | date:'short'}}&nbsp;&nbsp;
       <span class="caret"></span>
     </button>
     <ul style="cursor:pointer;" ngbDropdownMenu class="dropdown-menu" aria-labelledby="simple-dropdown">
       <li *ngFor="let row of versions">
-        <button class="dropdown-item" (click)="selectVersion(row)">{{row.number}} &nbsp;&nbsp;&#8208;&nbsp;&nbsp;
-          {{row.creationDate | date:'short'}} </button>
+        <button class="dropdown-item" (click)="selectVersion(row)">{{row.number}} &nbsp;&nbsp;&#8208;&nbsp;&nbsp; {{row.name}} &nbsp;&nbsp;&#8208;&nbsp;&nbsp; {{row.creationDate | date:'short'}} </button>
       </li>
       <li class="dropdown-divider"></li>
 
       <li class="dropdown-item">
         <button class="dropdown-item" (click)="openDialog(version)">Derive new version from current version
-          {{latestVersion.number}}</button>
+          {{version.number}}</button>
       </li>
     </ul>
 
   </span>
 
-  <div class="card-body">
-    {{version.comment}}
+  
+  <div class="card-body">   
 
     <br>
     <br>
     <div id="not-frozen" *ngIf="!version.frozen" style="height:28px">
-      <button class="btn btn-primary btn-sm btn-mrgn" mat-raised-button [disabled]="!version.privateStatus" (click)="addProject()">Add
+      <button class="btn btn-primary btn-sm btn-mrgn" mat-raised-button [disabled]="!version.privateStatus" (click)="addProject(version)">Add
         Project</button>
 
       <button mat-raised-button [disabled]="loading || saving || versions.length <= 1 || !version.privateStatus" class="btn btn-sm btn-danger btn-mrgn"
         (click)="removeVersion(version.id)">
         <i class="glyphicon glyphicon-trash"></i> Delete this version</button>
-      <button mat-raised-button [disabled]="loading || saving || !version.privateStatus" class="btn btn-sm btn-danger btn-mrgn"
+      <button mat-raised-button [disabled]="loading || saving || isPublic()" class="btn btn-sm btn-danger btn-mrgn"
```

Comment:
I noticed when you derive a version from another version and delete the parent version, the link is lost. Could you look into that also?

Reply:
right now, code changes have been made in the backend to throw an error if the parentversion doesnt exist in the DB. corresponding frontend changes have been done in this branch. but now you will have to use addexception branch in the backend for testing this.(This addexception branch is in sync with develop, only one LOC has been added, so will raise a PR for merging this. But meanwhile you can use this branch for testing this.